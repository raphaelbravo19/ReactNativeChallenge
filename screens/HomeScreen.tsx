import { useNavigation } from "@react-navigation/native";
import React, { useContext, useState } from "react";
import { StyleSheet, Text, View } from "react-native";
import { Button, Input } from "../components/ui";
import { Store } from "../context";
import { WHITE } from "../utils/colors";

export default function HomeScreen() {
  const [name, setName] = useState<string>("");
  const [password, setPassword] = useState<string>("");

  const { dispatch } = useContext(Store);

  const navigation = useNavigation();

  const onSignIn = () => {
    if (name === "" || password === "") {
      alert("Incorrect username or password");
      return;
    }
    dispatch({ type: "SIGNIN", payload: { username: name } });
    navigation.navigate("List");
  };

  return (
    <View style={styles.container}>
      <Text style={styles.title}>Welcome</Text>
      <Input
        value={name}
        onChangeText={setName}
        placeholder={"Enter your name"}
        containerStyle={styles.input}
      />
      <Input
        value={password}
        onChangeText={setPassword}
        placeholder={"Enter your password"}
        secureTextEntry
        containerStyle={styles.input}
      />
      <Button onPress={onSignIn} style={styles.button}>
        Sign in
      </Button>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: WHITE,
    alignItems: "stretch",
    justifyContent: "center",
    paddingHorizontal: 30,
  },
  title: {
    fontSize: 24,
    fontWeight: "bold",
    textAlign: "center",
    marginBottom: 28,
  },
  input: {
    marginTop: 16,
  },
  button: {
    marginTop: 44,
    marginHorizontal: 5,
    marginBottom: 200,
  },
});
