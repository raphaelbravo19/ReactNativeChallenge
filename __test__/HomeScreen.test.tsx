import React from "react";
import renderer from "react-test-renderer";
import HomeScreen from "../screens/HomeScreen";

test("home renders", () => {
  const tree = renderer.create(<HomeScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
