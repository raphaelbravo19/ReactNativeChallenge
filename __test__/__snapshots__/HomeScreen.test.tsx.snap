// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home renders 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 30,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 24,
        "fontWeight": "bold",
        "marginBottom": 28,
        "textAlign": "center",
      }
    }
  >
    Welcome
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#D1D5DB",
          "borderRadius": 6,
          "borderWidth": 1,
          "overflow": "hidden",
        },
        Object {
          "marginTop": 16,
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Enter your name"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "color": "#6B7280",
            "fontSize": 16,
            "fontWeight": "400",
            "paddingHorizontal": 13,
            "paddingVertical": 9,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#D1D5DB",
          "borderRadius": 6,
          "borderWidth": 1,
          "overflow": "hidden",
        },
        Object {
          "marginTop": 16,
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Enter your password"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Array [
          Object {
            "color": "#6B7280",
            "fontSize": 16,
            "fontWeight": "400",
            "paddingHorizontal": 13,
            "paddingVertical": 9,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1FC4DB",
        "borderRadius": 6,
        "marginBottom": 200,
        "marginHorizontal": 5,
        "marginTop": 44,
        "opacity": 1,
        "paddingVertical": 12,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 16,
          "fontWeight": "500",
        }
      }
    >
      Sign in
    </Text>
  </View>
</View>
`;
